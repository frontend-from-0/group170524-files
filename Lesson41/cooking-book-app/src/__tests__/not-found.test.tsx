import { render, screen } from "@testing-library/react";
import NotFound from "@/app/not-found";

describe("Page not found", () => {
  it("render text content correctly", () => {
    render(<NotFound />);

    const heading = screen.getByRole("heading", { level: 1 });
    const link = screen.getByTestId("link");

    expect(link).toHaveTextContent("Take me to the homepage");
    expect(link).toHaveAttribute("href", "/");

    expect(heading).toHaveTextContent("Page Not Found");
  });

  it("render image correctly", () => {
    render(<NotFound />);

    const image = screen.getByTestId("image");

    // TODO: replace src value with a regex
    expect(image).toHaveAttribute(
      "src",
      "/_next/image?url=%2Fconfused-chef.jpg&w=640&q=75",
    );
    expect(image).toHaveAttribute(
      "alt",
      "a confused chef looking at a cooking book",
    );
  });
  describe("Snapshot", () => {
    it("renders Not Found page unchanged", () => {
      const { container } = render(<NotFound />);
      expect(container).toMatchSnapshot();
    });
  })
});
