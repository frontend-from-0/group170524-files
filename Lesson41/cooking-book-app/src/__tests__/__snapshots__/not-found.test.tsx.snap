// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page not found Snapshot renders Not Found page unchanged 1`] = `
<div>
  <div
    class="text-center my-8"
  >
    <h1>
      Page Not Found
    </h1>
    <div>
      <img
        alt="a confused chef looking at a cooking book"
        class="mx-auto my-8"
        data-nimg="1"
        data-testid="image"
        decoding="async"
        height="300"
        loading="lazy"
        src="/_next/image?url=%2Fconfused-chef.jpg&w=640&q=75"
        srcset="/_next/image?url=%2Fconfused-chef.jpg&w=384&q=75 1x, /_next/image?url=%2Fconfused-chef.jpg&w=640&q=75 2x"
        style="color: transparent;"
        width="300"
      />
    </div>
    <a
      data-testid="link"
      href="/"
    >
      Take me to the homepage
    </a>
  </div>
</div>
`;
